text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,424 @@
+  tests for the target, using emulation if possible or dedicated hardware if
+  necessary.
+- The target must document its baseline expectations for the features or
+  versions of CPUs, operating systems, libraries, runtime environments, and
+  similar.
+- If introducing a new tier 2 or higher target that is identical to an existing
+  Rust target except for the baseline expectations for the features or versions
+  of CPUs, operating systems, libraries, runtime environments, and similar,
+  then the proposed target must document to the satisfaction of the approving
+  teams why the specific difference in baseline expectations provides
+  sufficient value to justify a separate target.
+  - Note that in some cases, based on the usage of existing targets within the
+    Rust community, Rust developers or a target's maintainers may wish to
+    modify the baseline expectations of a target, or split an existing target
+    into multiple targets with different baseline expectations. A proposal to
+    do so will be treated similarly to the analogous promotion, demotion, or
+    removal of a target, according to this policy, with the same team approvals
+    required.
+    - For instance, if an OS version has become obsolete and unsupported, a
+      target for that OS may raise its baseline expectations for OS version
+      (treated as though removing a target corresponding to the older
+      versions), or a target for that OS may split out support for older OS
+      versions into a lower-tier target (treated as though demoting a target
+      corresponding to the older versions, and requiring justification for a
+      new target at a lower tier for the older OS versions).
+- Tier 2 targets must not leave any significant portions of `core` or the
+  standard library `unimplemented!()`, unless they cannot possibly be supported
+  on the target.
+- If the target supports C code, the new Rust target should support the C
+  calling convention for the platform via `extern "C"`. The C calling
+  convention does not need to be the default Rust calling convention for the
+  target, however.
+- The target must build reliably in CI.
+- The approving teams may additionally require that a subset of tests pass in
+  CI, such as enough to build a functional "hello world" program, or equivalent
+  "smoke tests". In particular, this requirement may apply if the target builds
+  host tools, or if the tests in question provide substantial value via early
+  detection of critical problems.
+- Building the target in CI must not take substantially longer than the current
+  slowest target in CI. This requirement is subjective, to be evaluated by the
+  infrastructure team, and will take the community importance of the target
+  into account.
+- Tier 2 targets must support building on the existing CI infrastructure. In
+  particular, new tier 2 targets must support cross-compiling, and must not
+  require using the target as the host for builds.
```

Comment:
I'd move this to "should". I don't expect the infra team to grant an exception often (or ever), but for very important targets that might happen (for example if in a parallel universe it wasn't possible to cross-compile to AArch64 macOS).